protos/feast/core/DatastoreTable.proto

```diff
@@ -0,0 +1,43 @@
+//
+// * Copyright 2021 The Feast Authors
+// *
+// * Licensed under the Apache License, Version 2.0 (the "License");
+// * you may not use this file except in compliance with the License.
+// * You may obtain a copy of the License at
+// *
+// *     https://www.apache.org/licenses/LICENSE-2.0
+// *
+// * Unless required by applicable law or agreed to in writing, software
+// * distributed under the License is distributed on an "AS IS" BASIS,
+// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// * See the License for the specific language governing permissions and
+// * limitations under the License.
+//
+
+syntax = "proto3";
+
+package feast.core;
+option java_package = "feast.proto.core";
+option java_outer_classname = "DatastoreTableProto";
+option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";
+
+// Represents a Datastore table
+message DatastoreTable {
+    // Feast project of the table
+    string project = 1;
+
+    // Name of the table
+    string name = 2;
+
+    // GCP project id, which can be null
+    oneof project_id {
+        bool project_id_null = 3;
+        string project_id_value = 4;
+    }
+
+    // Datastore namespace, which can be null
+    oneof namespace {
+        bool namespace_null = 5;
+        string namespace_value = 6;
+    }
```

Comment:
These should probably be `google.protobuf.StringValue`. That's a struct type, that can be null - so you don't have to implement the same wrapper here and can reuse `StringValue` instead.